### PR TITLE
Improve the CVE feed python script HTTP request

### PR DESCRIPTION
**Pull request by mtardy**
I just wrote that in reaction to Pushkar's quick fix https://github.com/kubernetes/sig-security/pull/65, which is completely valid by the way (while the k8s CVEs number is <= 100), I just saw an area for improvement.

The previous implementation ignored all results that exceeded the number of items per page. This implementation uses the link HTTP header to iterate over potential next pages and should be a bit more robust.

If you want to try this to make sure it works on your computer, I suggest changing line 21 to:
```
state:closed+repo:kubernetes/kubernetes&per_page=20'
```
Be careful to not put something like `per_page=2` or Github will rate limit you notably because the request is unauthenticated!

/cc @PushkarJ @nehaLohia27

### CODE REVIEW

**Comment by Skybound1** [comment]
Reading through it looks good to me... not sure why you have the `.copy()` on the initial `gh_items` though

**Comment by mtardy** [comment]
> not sure why you have the `.copy()` on the initial `gh_items` though

To be honest I'm never writing Python and didn't want `gh_items` to point to an array that would be reassigned later (line 33). But it seems that `res.json()['items']` somehow already create a copy of the list. Do you know what I mean?

**Comment by Skybound1** [comment]
> > not sure why you have the `.copy()` on the initial `gh_items` though
> 
> To be honest I'm never writing Python and didn't want `gh_items` to point to an array that would be reassigned later (line 33). But it seems that `res.json()['items']` somehow already create a copy of the list. Do you know what I mean?

Ah fair, so from my understanding of Python, I don't think you need to worry. You never re-assign `gh_items` anywhere, at most you modify it with the `extend` function which will just append to the current list instead of re-assigning.

From the perspective of pointing to a list from `res` and that being changed underneath `gh_items`, you don't need to worry about that for two reasons. 1) The `.json()` call actually creates a separate object, so you never point to `res` in the first place (which is the bit you mentioned about it already creates a copy), 2) `res` gets re-assigned later as opposed to modified and so would point to a different object in memory, so if 1 wasn't the case `gh_items` would still point to the original version of `res` not the new one. The time you might be concerned is if `res` was directly the object that contained the list, and then you did list modifications _within_ `res` as opposed to just re-assigning it, which you're not doing here.

File: sig-security-tooling/cve-feed/hack/fetch-official-cve-feed.py
Comment (mtardy) [inline sig-security-tooling/cve-feed/hack/fetch-official-cve-feed.py:25]:
```suggestion
gh_items = res.json()['items']
```

**Comment by mtardy** [comment]
Thanks a lot, Skybound1 for the review :)

**Review by nehaLohia27 — APPROVED** [review]

**Comment by PushkarJ** [comment]
/label tide/merge-method-squash

(@mtardy Since #68 was merged, wanted to make sure this gets rebased on top of `main`)

**Comment by PushkarJ** [comment]
Thank you @Skybound1 for very informative review and @mtardy for this improvement that will serve us for the long term.

@mtardy let's hold this until EOD Sunday, so we know that #68 merge did not cause any issues. You can check if the job is green: https://testgrid.k8s.io/sig-security-cve-feed#auto-refreshing-official-cve-feed on Monday your time. If it is, feel free to unhold, so this gets merged :) 

/lgtm
/approve
/hold

**Pull request merged**
